### ISSUE TITLE
Add/remove tags and validator execution cycle

### ISSUE THREAD

**Issue by myConsciousness** [eyoeldefare/textfield_tags#34]
Hi thanks for this amazing development!

I tried to do a duplicate tag check in this widget with a validator, but it looks like the onTag process is running earlier than the validator process. I think the validator process should be run first, is there a purpose for this cycle?

**Comment by eyoeldefare** [comment]
Hey there, yeah duplicates are removed automatically from internal storage when they are entered, the validator mainly handles and validates the exact entered String. It has no memory of the past entered String tags. I did not think people wanted to have duplicate tags so I simply removed them, in the future, I think we can let people handle duplicates by them selves.

**Comment by myConsciousness** [comment]
Hello @eyoeldefare !

I noticed the process of automatically removing duplicate tags when I was looking at the source after publishing this issue. I had implemented a duplicate tag checker myself and was trying to output an error message as a validator result if there were duplicates.

I checked the GitHub tag feature, and it seems that there are controls in place to prevent tagging of duplicates, although they are not automatically deleted. No error message was output.

I think the automatic removal of duplicate tags is a good feature. However, depending on the application, there may be people like me who would like to get an error message if there are duplicate tags :)
